File: pricing-template/src/main/java/com/caplin/template/PricingAdapterTemplate.java
package com.caplin.template;

import com.caplin.datasource.DataSource;
import com.caplin.template.channel.ChannelTemplate;
import com.caplin.template.channel.JsonChannelTemplate;
import com.caplin.template.container.ContainerTemplateDataProvider;
import com.caplin.template.pricing.PricingTemplateDataProvider;
import com.caplin.template.type2.Type2DataProvider;

public class PricingAdapterTemplate {

    private final DataSource dataSource;

    public PricingAdapterTemplate(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    public void initialise() {
        PricingTemplateDataProvider pricingTemplateDataProvider = new PricingTemplateDataProvider(dataSource);
        ContainerTemplateDataProvider containerTemplateDataProvider = new ContainerTemplateDataProvider(dataSource);
        ChannelTemplate channelTemplate = new ChannelTemplate(dataSource);
        JsonChannelTemplate jsonChannelTemplate = new JsonChannelTemplate(dataSource);
        Type2DataProvider type2DataProvider = new Type2DataProvider(dataSource);

        pricingTemplateDataProvider.initialise();
        containerTemplateDataProvider.initialise();
        channelTemplate.initialise();
        jsonChannelTemplate.initialise();
        type2DataProvider.initialise();
    }

    public static void main(String[] args) {
        DataSource dataSource = DataSource.fromArgs(args);

        PricingAdapterTemplate pricingAdapterTemplate = new PricingAdapterTemplate(dataSource);
        pricingAdapterTemplate.initialise();

        dataSource.start();
    }
}

File: pricing-template-dotnet/src/PricingAdapterTemplate.cs
using Caplin.DataSource;
using Caplin.Logging;

using System.Threading;

using Caplin.Template;

namespace Caplin.Template
{
    public class PricingAdapterTemplate
    {
        private IDataSource dataSource;

        public PricingAdapterTemplate(IDataSource dataSource)
        {
            this.dataSource = dataSource;
        }

        public void Initialise()
        {
            var pricingTemplateDataProvider = new PricingTemplateDataProvider(dataSource);
            //ContainerTemplateDataProvider containerTemplateDataProvider = new ContainerTemplateDataProvider(dataSource);
            var channelTemplate = new ChannelTemplate(dataSource);
            //Type2DataProvider type2DataProvider = new Type2DataProvider(dataSource);

            pricingTemplateDataProvider.Initialise();
            //containerTemplateDataProvider.initialise();
            channelTemplate.Initialise();
            //type2DataProvider.initialise();
        }

        static void Main(string[] args)
        {

            ILogger logger = new ConsoleLogger();
            IDataSource dataSource = new Caplin.DataSource.DataSource("demosource.conf", args, logger);
            var templateAdapter = new PricingAdapterTemplate(dataSource);

            templateAdapter.Initialise();

            dataSource.Start();

            // Now we have to loop forever - DataSource.NET runs in a background thread
            while (true)
            {
                Thread.Sleep(1000);
            }
        }
    }

}
